src/renderer/components/dialog/publicationInfos/opdsControls.tsx

```diff
@@ -62,4 +64,11 @@ const mapDispatchToProps = (dispatch: TDispatch) => {
     };
 };
 
-export default connect(undefined, mapDispatchToProps)(withTranslator(OpdsControls));
+// any because recursive type doesn't works
+const mapStateToProps = (state: RootState, props: any) => {
+    return {
+        buttonIsDisabled: state.download.downloads.findIndex((pub) => pub.url === props.publication.url) > -1,
```

Comment:
I think `state.download.downloads.findIndex((pub) => pub.url === props.publication.url) > -1` can be simplified:
`state.download.downloads.find((pub) => pub.url === props.publication.url)`

Reply:
Mmmh, maybe not :)
`buttonIsDisabled` is a boolean, so better not assign an object. Scrap this comment!